pkg/apis/autoscaling/v1/fleetautoscaler.go

```diff
@@ -328,6 +400,130 @@ func (b *BufferPolicy) ValidateBufferPolicy(causes []metav1.StatusCause) []metav
 	return causes
 }
 
+// ValidateCounterPolicy validates the FleetAutoscaler Counter policy settings
+// #TODO: Does NOT validate if a Counter with name CounterPolicy.Key is present in the fleet.
+// nolint:dupl  // Linter errors on lines are duplicate of ValidateListPolicy
+func (c *CounterPolicy) ValidateCounterPolicy(causes []metav1.StatusCause) []metav1.StatusCause {
```

Comment:
I think Key should be required because I can't think of an use case where user wants to scale based on all counters. WDYT?

Reply:
Agreed it should be required -- I didn't see a way of validating here that the key actually exists in the fleet since the fleet isn't passed through into the fleetautoscaler.go.

Reply:
Make sense. I think we can make it a no-op in controller if the fleet doesn't have the key.